hasura/migrations/default/1704993161810_create_table_app_metadata/up.sql

```diff
@@ -134,6 +134,23 @@ END;
 $$ LANGUAGE plpgsql;
 
 CREATE TRIGGER "trigger_enforce_app_id_row_limit"
+AFTER INSERT OR UPDATE ON "public"."app_metadata"
+FOR EACH ROW
+EXECUTE FUNCTION enforce_app_id_row_limit();
+
+CREATE FUNCTION set_unique_verification_status_row()
```

Comment:
how about using a trigger like you're suggesting but without having an additional field. we can reject changes that would result in the trigger constraint being violated. this way we can also get more useful error messages